flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/AbstractFlinkService.java

```diff
@@ -234,23 +235,29 @@ public JobID submitJobToSessionCluster(
 
     @Override
     public boolean isJobManagerPortReady(Configuration config) {
-        final URI uri;
+        SocketAddress socketAddress;
         try (var clusterClient = getClusterClient(config)) {
-            uri = URI.create(clusterClient.getWebInterfaceURL());
+            socketAddress = getSocketAddress(clusterClient);
+
         } catch (Exception ex) {
             throw new FlinkRuntimeException(ex);
         }
-        SocketAddress socketAddress = new InetSocketAddress(uri.getHost(), uri.getPort());
-        Socket socket = new Socket();
-        try {
+
+        try (Socket socket = new Socket()) {
             socket.connect(socketAddress, 1000);
-            socket.close();
             return true;
         } catch (IOException e) {
             return false;
         }
     }
 
+    protected SocketAddress getSocketAddress(RestClusterClient<String> clusterClient)
+            throws MalformedURLException {
+        final URL url = new URL(clusterClient.getWebInterfaceURL());
```

Comment:
Flink uses URL to create this String. Hence using URL to extract the hostname makes sense. URI fails to interpret the host name correctly in certain URLs like `http://test.01-bla:8081` and treats it as a different part of an URI.